client/components/proposal/ProposalActionAddModal.tsx

```diff
@@ -36,6 +48,18 @@ export const ProposalActionAddModal = ({
         </ul>
         {step === 0 && (
           <AddActionContractForm
+            onChange={(data) => {
+              setIsProxy(data.name.includes('Proxy'));
```

Comment:
I think if we check the contract abi for an `implementation()` method, call that, and search the contracts for an address matching the result of that call. If it doesn't find anything we should display an error (e.g. Contract is a proxy but no implementation found for <address>).

Reply:
This makes sense. Have introduced a mainnet call to grab the implementation address of all proxy contracts (those with an `implementation()` method), alongside loading UI and a relevant error message should no ABI matching the fetched address be present in our ABI list.